File: seed/php-model/trace/src/V2/Problem/VoidFunctionDefinition.php
<?php

namespace Seed\V2\Problem;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;
use Seed\Core\ArrayType;

class VoidFunctionDefinition extends SerializableType
{
    /**
     * @var array<Parameter> $parameters
     */
    #[JsonProperty("parameters"), ArrayType([Parameter::class])]
    public array $parameters;

    /**
     * @var FunctionImplementationForMultipleLanguages $code
     */
    #[JsonProperty("code")]
    public FunctionImplementationForMultipleLanguages $code;

    /**
     * @param array<Parameter> $parameters
     * @param FunctionImplementationForMultipleLanguages $code
     */
    public function __construct(
        array $parameters,
        FunctionImplementationForMultipleLanguages $code,
    ) {
        $this->parameters = $parameters;
        $this->code = $code;
    }
}

File: seed/pydantic-v2/trace/resources/v_2/resources/v_3/resources/problem/types/void_function_definition.py
from pydantic import BaseModel
from typing import List
from resources.v_2.resources.v_3.resources.problem.types.parameter import Parameter
from resources.v_2.resources.v_3.resources.problem.types.function_implementation_for_multiple_languages import FunctionImplementationForMultipleLanguages
from dt import datetime
from core.datetime_utils import serialize_datetime
class VoidFunctionDefinition(BaseModel):
    parameters: List[Parameter]
    code: FunctionImplementationForMultipleLanguages
    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

